### PR TITLE
Add blob support to DBIter

### PR DESCRIPTION
**Pull request by ltamasi**
Summary:
The patch adds iterator support to the integrated BlobDB implementation.
Whenever a blob reference is encountered during iteration, the corresponding
blob is retrieved by calling `Version::GetBlob`, assuming the `expose_blob_index`
(formerly `allow_blob`) flag is *not* set. (Note: the flag is set by the old stacked
BlobDB implementation, which has its own blob file handling/blob retrieval logic.)

In addition, `DBIter` now uniformly returns `Status::NotSupported` with the error
message `"BlobDB does not support merge operator."` when encountering a
blob reference while performing a merge (instead of potentially returning a
message that implies the database should be opened using the stacked BlobDB's
`Open`.)

TODO: We can implement support for lazily retrieving the blob value (or in other
words, bypassing the retrieval of blob values based on key) by extending the `Iterator`
API with a new `PrepareValue` method (similarly to `InternalIterator`, which already
supports lazy values).

Test Plan:
`make check`

### CODE REVIEW

**Review by riversand963 — COMMENTED** [review]
Thanks @ltamasi for the PR. Left a few minor comments.

File: db/db_iter.cc [hunk 1]
```diff
@@ -342,9 +342,17 @@ bool DBIter::FindNextUserEntryInternal(bool skipping_saved_key,
           case kTypeBlobIndex:
             if (start_seqnum_ > 0) {
               if (ikey_.sequence >= start_seqnum_) {
-                assert(ikey_.type != kTypeBlobIndex);
                 saved_key_.SetInternalKey(ikey_);
-                valid_ = true;  // TODO???
+
+                if (ikey_.type == kTypeBlobIndex) {
+                  if (SetBlobValueIfNeeded(ikey_.user_key, iter_.value())) {
```
[the remark this answers is not in the document]
Comment (ltamasi) [reply db/db_iter.cc:360]: 😳 Great catch, thanks @riversand963 !

File: db/db_iter.h [hunk 2]
```diff
@@ -161,15 +161,22 @@ class DBIter final : public Iterator {
...
+    }
+
     if (current_entry_is_merged_) {
       // If pinned_value_ is set then the result of merge operator is one of
       // the merge operands and we should return it.
       return pinned_value_.data() ? pinned_value_ : saved_value_;
-    } else if (direction_ == kReverse) {
+    }
+
+    if (direction_ == kReverse) {
```
[the remark this answers is not in the document]
Comment (ltamasi) [reply db/db_iter.h:175]: Sure. Matter of taste I guess. :)

**Review by riversand963 — APPROVED** [review]
LGTM.

**Pull request closed without merging**